src/util/path/path.ts

```diff
@@ -560,20 +652,23 @@ const pathIterator = (
  * @param {Number} distance from starting point, in pixels.
  * @return {Object} info object with x and y ( the point on canvas ) and angle, the tangent on that point;
  */
-const findPercentageForDistance = (segInfo: CurveInfo, distance: number) => {
+const findPercentageForDistance = (
+  segInfo: TCurveInfo<'Q' | 'C'>,
+  distance: number
+): TPointAngle => {
   let perc = 0,
     tmpLen = 0,
     tempP = { x: segInfo.x, y: segInfo.y },
-    p,
+    p: TPointAngle = { ...tempP, angle: 0 },
     nextLen,
     nextStep = 0.01,
-    lastPerc;
+    lastPerc = 0;
   // nextStep > 0.0001 covers 0.00015625 that 1/64th of 1/100
   // the path
   const iterator = segInfo.iterator,
     angleFinder = segInfo.angleFinder;
   while (tmpLen < distance && nextStep > 0.0001) {
-    p = iterator(perc);
+    p = { ...iterator(perc), angle: 0 };
```

Comment:
can you explain this?

Reply:
I see that in line https://github.com/fabricjs/fabric.js/pull/8787/files#diff-d309b6b9d88030a0c982bdc0536f31a874f2441c726e9dc7f293ff2dfbed2992R685 it is set 
Is it that why?

Reply:
I wanted to return `p` but I couldn't set `p` to `iterator` there because it only returned a `Point` and not a `TPointAngle`. I was just trying to make ts happy

Reply:
So the same applies here.
Making a crazy thing happy is to be avoided 😵

Reply:
OK I went ahead and changed it. I realized I wasn't thinking when I was doing this one lol, it's much simpler now.